File: answers/other/in sap/Jump Game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        
        reach = 1
        for i in nums:
            if reach <= 0:
                return False
            else:
                reach = max(reach - 1, i)
        return True

File: answers/leetcode/Jump Game/Jump Game.cpp
//@type greedy, only one loop
class Solution {
public:
    bool canJump(int A[], int n) {
		int can_jump = 0;
		for (int i = 0; i <= can_jump; ++ i)
		{
			can_jump = std::max(can_jump, i + A[i]);
			if (can_jump >= n - 1)
			{
				return true;
			}
		}
		return false;
    }
};
